clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -92,11 +92,37 @@ public synchronized Set<TopicPartition> assignment() {
         return this.subscriptions.assignedPartitions();
     }
 
-    /** Simulate a rebalance event. */
+    /**
+     * Simulate a rebalance event.
+     */
     public synchronized void rebalance(Collection<TopicPartition> newAssignment) {
-        // TODO: Rebalance callbacks
+        // prepare for rebalance callback
+        Set<TopicPartition> oldAssignmentSet = this.subscriptions.assignedPartitions();
+        Set<TopicPartition> newAssignmentSet = new HashSet<>(newAssignment);
+        List<TopicPartition> added = new ArrayList<>(newAssignment.size());
```

Comment:
I'm pretty sure you can do set.removeAll(stuff) - saves some code there.

Reply:
i changed it to streams and reduced it to one line per list